i6_models/parts/conformer/convolution.py

```diff
@@ -85,3 +90,78 @@ def forward(self, tensor: torch.Tensor) -> torch.Tensor:
         tensor = self.pointwise_conv2(tensor)
 
         return self.dropout(tensor)
+
+
+@dataclass
+class ConformerConvolutionV2Config(ConformerConvolutionV1Config):
+    """
+    New attribute:
+        dropout_broadcast_axes: string of axes to which dropout is broadcast, e.g. "T" for broadcasting to the time axis
+                                setting to None to disable broadcasting
+    Allows even kernel size
+    """
+
+    dropout_broadcast_axes: Optional[str] = None
+
+    def check_valid(self):
+        assert self.kernel_size % 2 == 1, "ConformerConvolutionV1 only supports odd kernel sizes"
+
+        assert self.dropout_broadcast_axes is None or self.dropout_broadcast_axes in [
+            "B",
+            "T",
+            "BT",
+        ], "invalid value, supported are None, 'B', 'T' and 'BT'"
+
+
+class ConformerConvolutionV2(ConformerConvolutionV1):
+    """
+    Augments ConformerMHSAV1 with dropout broadcasting
+    """
+
+    def __init__(self, model_cfg: ConformerConvolutionV2Config):
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__(model_cfg)
+
+        self.dropout_broadcast_axes = model_cfg.dropout_broadcast_axes
+        self.dropout = (
+            nn.Dropout1d(model_cfg.dropout) if model_cfg.dropout_broadcast_axes else nn.Dropout(model_cfg.dropout)
```

Comment:
Not a strong issue, but why do you have dropout here (in convolution) as explicit module but not for Feed-Forward and MHSA? I think this should be consistent. @michelwi @christophmluscher @curufinwe I would prefer to have it explicit, because then it is printed when you print the model.

Reply:
> Not a strong issue, but why do you have dropout here (in convolution) as explicit module but not for Feed-Forward and MHSA? I think this should be consistent. @michelwi @christophmluscher @curufinwe I would prefer to have it explicit, because then it is printed when you print the model.

I just followed the V1 implementations. But I could make them all modules. Anyone against that?

Reply:
I am OK with the suggested change. It would improve readability :)